### PR TITLE
Add PlusHalf wrapper type

### PR DESCRIPTION
**Pull request by simonbyrne**
This should help us avoid off-by-one error when writing staggered stencils.

### CODE REVIEW

File: src/Operators/finitedifference.jl [hunk 1]
```diff
@@ -120,6 +120,54 @@ interior_indices1 = 1:n
...
+right_face_boundary_idx(arg) = right_face_boundary_idx(axes(arg))
+left_center_boundary_idx(arg) = left_center_boundary_idx(axes(arg))
+right_center_boundary_idx(arg) = right_center_boundary_idx(axes(arg))
+
+Δh_f2f(space::Spaces.FiniteDifferenceSpace, idx::Integer) = space.Δh_f2f[idx]
+Δh_c2c(space::Spaces.FiniteDifferenceSpace, idx::PlusHalf) =
+    space.Δh_c2c[idx.i + 1]
+
+Δh_left_bf2c(space::Spaces.FiniteDifferenceSpace) = space.Δh_c2c[1]
+Δh_right_c2bf(space::Spaces.FiniteDifferenceSpace) = space.Δh_c2c[end]
```
Comment (simonbyrne) [inline src/Operators/finitedifference.jl:170]: this is why we need `lastindex` @jakebolewski

**Review by jakebolewski — APPROVED** [review]
:100:  / 2 !

**Pull request merged**